Program.cs

```diff
@@ -3,17 +3,132 @@
+               Write ($"│ {(i == colPos ? "♕" : " ")} ");
+            WriteLine ("│");
+         }
+      }
+
+      /// <summary>Checks whether the solution is unique or not</summary>
+      /// <param name="a">solution array</param>
+      /// <returns>Returns true if the solution is unique</returns>
+      static bool IsUnique (int[] a) {
+         if (!IsMirror (a) && !IsRotatedImage (a)) return true;
+         return false;
+
+         /// <summary>Checks whether the solution is a mirror image of the previous solutions</summary>
+         static bool IsMirror (int[] a) => IsDuplicate (a.Reverse ().ToArray ());
+
+         /// <summary>Checks whether the solution is identical to the previous solutions</summary>
+         static bool IsDuplicate (int[] a) => sAllSlns.Any (x => x.SequenceEqual (a));
+
+         /// <summary>Checks whether the solution is a rotated image of the previous solutions</summary>
+         static bool IsRotatedImage (int[] a) {
+            int[] b = new int[8], c = new int[8];
+            CopySolution (a, c);
+            for (int i = 0; i < 3; i++) {
+               Rotate90 (b, c);
+               if (IsMirror (b) || IsDuplicate (b)) return true;
+            }
+            return false;
+
+            /// <summary>Rotates the solution by 90 degrees</summary>
+            static void Rotate90 (int[] b, int[] c) {
+               for (int i = 0; i < 8; i++) b[c[i]] = 7 - i;
+               CopySolution (b, c);
+            }
+         }
+      }
+
+      /// <summary>Checks whether the given position is safe to place the queen</summary>
+      /// <param name="row">Row value of the position to be checked</param>
+      /// <param name="col">Column value of the position to be checked</param>
+      /// <returns>Returns true if the position is safe</returns>
+      static bool IsSafe (int row, int col) {
+         for (int prevRow = 0; prevRow < row; prevRow++) {
+            var prevCol = sColumns[prevRow];
+            if (col == prevCol ||
+               Math.Abs (row - prevRow) == Math.Abs (col - prevCol)) return false;
+         }
+         return true;
+      }
+
+      /// <summary>Stores the safe positions of a solution in an array
+      /// and stores all the solution arrays in a list</summary>
+      /// <param name="row">Fixing the value of row to go through all columns of that row</param>
+      static void PlaceQueen (bool isUnique, int row) {
+         for (int col = 0; col < 8; col++) {
+            if (IsSafe (row, col)) {
+               sColumns[row] = col;
+               var sln = sColumns.ToArray ();
+               if (isUnique ? row == 7 && IsUnique (sln) : row == 7) sAllSlns.Add (sln);
+               else PlaceQueen (isUnique, row + 1);
+            }
+         }
+      }
+
+      /// <summary>Creates a copy of the given array</summary>
+      static void CopySolution (int[] original, int[] copy) {
+         for (int i = 0; i < 8; i++) copy[i] = original[i];

```

Comment:
Can be simplified as 'static void CopySolution (int[] original, int[] copy) => Array.Copy (original, copy, 8);'